test/cloudwatchlogs/publish_logs_test.go

```diff
@@ -113,6 +113,35 @@ func TestWriteLogsToCloudWatch(t *testing.T) {
 	}
 }
 
+// TestAutoRemovalStopAgent configures agent to monitor a file with auto removal on.
+// Then it restarts the agent.
+// Verify the file is NOT removed.
+func TestAutoRemovalStopAgent(t *testing.T) {
```

Comment:
Just to sanity check the test, to make sure you got the config right (i.e. the auto_removal logic is actually being exercised), can you add the next scenario where you close the file and open `test.log2`, then make sure `test.log1` *is* deleted?

Reply:
Yes, I plan on adding a separate test case that verifies files GET removed.

Something like:
```
// TestAutoRemovalFileRotation repeatedly created files matching the monitored pattern.
// After creating each file, write some log lines, sleep and verify previous_file was auto removed.
// Does not verify log event is retrievable from CWL web service.
```

We do have unit tests for the auto_removal feature too.

Any test involving stopping/starting/restarting the agent though should be done as in integration test.

Reply:
OK, makes sense.  Are you doing that in this PR or another one?  I would also recommend combining them into one test case, again just to sanity check that the config was right.